typescript/packages/sphere-viewer/src/index.ts

```diff
@@ -0,0 +1,58 @@
+import './sphere-viewer.js';
+import { openSphere, connectToNoosphere, openFile } from './state/actions.js';
+import { store } from './state/store.js';
+
+(self as any).store = store;
+
+function getQueryState(): { [index: string]: string } {
+  const query = self.location.search.substring(1);
+  return query
+    .split('&')
+    .map((part) => {
+      return part.split('=').map((subPart) => decodeURIComponent(subPart));
+    })
+    .reduce((map: { [index: string]: string }, entry) => {
+      if (entry[0] && entry[1]) {
+        map[entry[0]] = entry[1];
+      }
+      return map;
+    }, {});
+}
+
+const queryState = getQueryState();
```

Comment:
Optional simplification idea:

```suggestion
const queryState = Object.fromEntries(new URLSearchParams(location.search));
```